base/src/main/java/org/aya/tyck/LittleTyper.java

```diff
@@ -73,10 +75,20 @@ public record LittleTyper(@NotNull TyckState state, @NotNull LocalCtx localCtx)
         yield piRaw instanceof FormTerm.Pi pi ? pi.substBody(app.arg().term()) : ErrorTerm.typeOf(app);
       }
       case FormTerm.Univ univ -> new FormTerm.Univ(univ.lift() + 1);
-      case FormTerm.Interval interval -> new FormTerm.Univ(0);
-      case PrimTerm.End end -> FormTerm.Interval.INSTANCE;
+      case FormTerm.Interval interval -> FormTerm.Univ.ZERO;
+      case PrimTerm.Mula end -> FormTerm.Interval.INSTANCE;
       case PrimTerm.Str str -> state.primFactory().getCall(PrimDef.ID.STR);
       case LitTerm.ShapedInt shaped -> shaped.type();
+      case FormTerm.PartTy ty -> FormTerm.Univ.ZERO;
+      case IntroTerm.HappyPartEl el -> {
```

Comment:
Hmm, should we store `A` in the partial term?

Reply:
It can happen that we are unable to infer the type `A`.

Reply:
> It can happen that we are unable to infer the type `A`.

I guess this only happens when a partial element doesn't have clauses? Currently we don't allow this case so maybe it's not needed yet?

Reply:
What if we have no clause in the partial at all?

Reply:
Then its type `Partial A r` can never reduce to `A` since we don't know if `isOne r` holds.

Reply:
Oh it seems agda allows empty partials

Reply:
I actually think empty partials are harmful. But we should store the type anyway, bc the boundaries can have different types